chat-widget/src/components/webchatcontainerstateful/webchatcontroller/middlewares/renderingmiddlewares/activityMiddleware.tsx

```diff
@@ -82,6 +88,48 @@ export const createActivityMiddleware = (renderMarkdown: (text: string) => strin
             return handleSystemMessage(next, args, card, renderMarkdown, systemMessageStyleProps);
         }
 
+        if (isTagIncluded(card, Constants.persistentChatHistoryMessagePullTriggerTag)) {
+            
+            // Safety check: if this is a new chat session and flag is false, auto-correct it
+            if (!LazyLoadHandler.hasMoreHistoryAvailable) {
+                LazyLoadHandler.setHasMoreHistoryAvailable(true);
+            }
+            
+            const receivedAt = card?.activity?.channelData?.webChat?.receivedAt;
```

Comment:
Can this be undefined or is it always a number? If always a number, disregard comment.

If can be undefined, the below will always return false. Which is also okay if that is what you want. Just thought to point it out.

Reply:
is always present, is the timestamp from webchat